code/mission/missionbriefcommon.cpp

```diff
@@ -609,7 +609,8 @@ void brief_preload_icon_anim(brief_icon *bi)
 	// force read of data from disk, so we don't glitch on initial playback
 	if ( ga->first_frame == -1 ) {
 		ga->first_frame = bm_load_animation(ga->filename, &ga->num_frames);
-		Assert(ga->first_frame >= 0);
+		if (ga->first_frame < 0)
+			Error(LOCATION, "Failed to load icon %s!", ga->filename);
```

Comment:
Good idea, but maybe *this* would be appropriate as a Warning instead of an Error.  Errors generally mean that the error is unrecoverable.  I speculate that the briefing animation code could be modified to accommodate an invalid icon without too much trouble.

Reply:
I wanted to keep the semantic of the previous assert with a better error message.
The briefing animation currently checks for `-1` in `first_frame` and just `Int3`s in that case. I guess that could be changed but I'm not sure if that's worth the effort since then you'll just get a briefing screen with a lot of missing icons.

I just noticed that there are a lot of similar `Assert`s in this file. I probably should change them as well.